File: elements-of-programming-interviews/2023/python/count_bits.py
from test_framework import generic_test


def count_bits(x: int) -> int:
    bits = 0
    while (x):
        if (x & 1):
            bits += 1
        x >>= 1
    return bits


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('count_bits.py', 'count_bits.tsv',
                                       count_bits))

File: elements-of-programming-interviews/2023/cpp/count_bits.cc
#include "test_framework/generic_test.h"

short CountBits(unsigned int x) {
  short count = 0;
  for (int i = 0; i < sizeof(int) * 8; i++) {
    if (x & 1) {
      count++;
    }
    x >>= 1;
  }
  return count;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"x"};
  return GenericTestMain(args, "count_bits.cc", "count_bits.tsv", &CountBits,
                         DefaultComparator{}, param_names);
}
